tests/ordering/tests.py

```diff
@@ -307,3 +307,11 @@ def test_order_by_f_expression_duplicates(self):
             ],
             attrgetter("headline")
         )
+
+    def test_related_ordering_duplicate_table_reference(self):
+        """
+        When an ordering references a model with an ordering referencing the same model
+        multiple time no infinite loop errors should be raised.
+        Refs #24654.
+        """
+        self.assertFalse(Reference.objects.all())
```

Comment:
It might be nice to create some objects and verify the ordering so that models aren't inadvertently refactored and ordering attributes lost which would eliminate their regression nature.